_posts/2022-09-08-storage-write-removal-before-conditional-termination.md

```diff
@@ -37,6 +37,8 @@ More concretely, the bug can occur if a contract contains the following pattern:
     - It overwrites the storage write in (1).
     - It reverts.
 
+If the initial storage write could possibly be read back (directly or in any internal or external call) between (1) and (3), it is, of course, never removed.
+
```

Comment:
Hm... thinking about this: While this is a nice thing to mention, it opens up another issue...

There can be cases, in which reading the initial storage value back can be resolved by the ``LoadResolver``... after which the bug can *still* kick in... so claiming that any read-back prevents the bug is also misloading... i.e.

```
{
  let x := 0
  let y := 1
  sstore(x, y)
  let z := sload(x)
  conditionallyReturn()
  sstore(x,y)
  ...
}
```
Can first load-resolve to
```
{
  let x := 0
  let y := 1
  sstore(x, y)
  let z := y
  conditionallyReturn()
  sstore(x,y)
  ...
}
```
and then *still* loose the first ``sstore``...

Reply:
I could remark that as well as a proviso to the proviso :-)... but this makes me wonder if *not* mentioning it isn't actually simpler, even though it may make this seem even more severe than it is...

Reply:
I think it is a good idea to mention it. I'd simply add something like this at the end of the sentence you added:
```
Note, however, that in presence of the optimizer it is not always easy to determine if a storage read at the Solidity level will actually translate to a load instruction in assembly. For example the [Load Resolver](https://docs.soliditylang.org/en/latest/internals/optimizer.html#load-resolver) step can use the knowledge of values that were written before by the contract to replace any `sload()` directly with the value that it would read.
```

Reply:
We should have footnotes in these posts :-).

Reply:
What about footnotes to footnotes :)

Reply:
> What about footnotes to footnotes :)

Even better :-D.